### PR TITLE
net-misc/minidlna: Refactor OpenRC runscript

### PR DESCRIPTION
**Pull request by jirutka**

### CODE REVIEW

**Comment by mgorny** [comment]
@xmw

File: net-misc/minidlna/files/minidlna-1.1.4.initd [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+	need net
+}
+
+start_pre() {
+	if [ "$M_USER" == "root" ]; then
+		eerror "It's not a good idea to run minidlna as root!"
+		exit 2
+	fi
+
+	if [ "${RESCAN/true/yes}" == "yes" ]; then
```
Comment (mgorny) [inline net-misc/minidlna/files/minidlna-1.1.4.initd]: Hmm, sorry for noticing this this late. Can't you use `yesno` here?

Comment (mgorny) [inline net-misc/minidlna/files/minidlna-1.1.4.initd]: Also `==` is non-POSIX.

File: net-misc/minidlna/files/minidlna-1.1.4.initd [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+		command_args+=" -R"
```
Comment (mgorny) [inline net-misc/minidlna/files/minidlna-1.1.4.initd]: `+=` is non-POSIX. You need to use plain `=`.

File: net-misc/minidlna/minidlna-1.1.4.ebuild [hunk 3]
```diff
@@ -72,7 +72,7 @@ src_install() {
 	doins ${PN}.conf
 
 	newconfd "${FILESDIR}"/${PN}-1.0.25.confd ${PN}
-	newinitd "${FILESDIR}"/${PN}-1.1.2.initd ${PN}
+	newinitd "${FILESDIR}"/${PN}-1.1.4.initd ${PN}
```
Comment (mgorny) [inline net-misc/minidlna/minidlna-1.1.4.ebuild]: I'd say this kind of change deserves a revbump.

**Comment by jirutka** [comment]
@mgorny I fixed issues you’ve mentioned and updated the commit.

**Comment by mgorny** [comment]
@xmw?

**Comment by mgorny** [comment]
@gentoo/openrc, could you take a second look at this, please?

**Comment by jirutka** [comment]
gentle ping

**Comment by jirutka** [comment]
Come on, this is opened for half a year. :crying_cat_face:

File: net-misc/minidlna/minidlna-1.1.4-r1.ebuild [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+inherit eutils linux-info systemd toolchain-funcs user
+
+DESCRIPTION="DLNA/UPnP-AV compliant media server"
+HOMEPAGE="http://minidlna.sourceforge.net/"
+SRC_URI="mirror://sourceforge/${PN}/${PV}/${P}.tar.gz
+	https://dev.gentoo.org/~xmw/${PN}-gentoo-artwork.patch.xz"
+
+LICENSE="BSD GPL-2"
+SLOT="0"
+KEYWORDS="amd64 arm x86"
```
Comment (Amynka) [inline net-misc/minidlna/minidlna-1.1.4-r1.ebuild]: You cannot revbump stable version to stable version. 

Meaning you have to change KEYWORDS="~amd64 ~arm ~x86"

Comment (mgorny) [inline net-misc/minidlna/minidlna-1.1.4-r1.ebuild]: This is not actually true. It's allowed if the change is simple and safe enough.

Comment (Amynka) [inline net-misc/minidlna/minidlna-1.1.4-r1.ebuild]: And is this simple and safe enough?

Comment (jirutka) [inline net-misc/minidlna/minidlna-1.1.4-r1.ebuild]: I’ve just rebased this PR to the current master. There’s minidlna-1.1.5 which is currently unstable, so I’ve created minidlna-1.1.5-r1 with the refactored runscript.

Comment (Amynka) [inline net-misc/minidlna/minidlna-1.1.4-r1.ebuild]: Can you please also update copyright? I will try to ping @xmw to look at it.

Comment (jirutka) [inline net-misc/minidlna/minidlna-1.1.4-r1.ebuild]: Ah, sorry, it’s fixed now.

**Comment by jirutka** [comment]
@xmw Why did you closed this PR?

**Comment by xmw** [comment]
Because i'm the maintainer of the package and adding the changes to the main tree just now.

**Comment by jirutka** [comment]
Ah, okay, thanks. It’s better to merge the changes and _after_ that close the PR.

**Comment by xmw** [comment]
I disagree on the fatal warning if run as root. I'm changing that to a non-fatal one.

**Comment by jirutka** [comment]
Hm, I can’t remember why I added this check; I usually don’t check if running as root in runscripts.

**Pull request closed without merging**